pysrc/utils.py

```diff
@@ -68,3 +82,189 @@ def is_string(obj):
         return isinstance(obj, basestring)
     else:
         return isinstance(obj, str)
+
+
+def remove_optional_dependencies(package_name):
+    """Removes optional dependencies/arbitrary identifiers from package name.
+
+    It looks at the end of the package name and searches for a pair of square
+    brackets with anything inside of them and replaces it with nothing.
+
+    Args:
+        package_name (str): package name to be processed
+
+    Examples:
+        opentelemetry-distro[otlp] -> opentelemetry-distro
+
+    Returns:
+        str: `package_name` without optional_dependencies/arbitrary identifiers
+    """
+    package_name = re.sub(r'\[[^]]*]$', '', package_name)
```

Comment:
This part of the code, as @JamesPatrickGill mentioned in a previous PR, is linear. I would have to rethink the way I process the data in order to not iterate over the dependencies. But I'm planning to do this when refactoring the pysrc.